en/dev_setup/dev_env_linux_arch.md

```diff
@@ -7,7 +7,7 @@ See [Toolchain Installation](../dev_setup/dev_env.md) for information about the
 
 The PX4-Autopilot repository provides a convenient script to set your Arch installation up for PX4 development: [Tools/setup/arch.sh](https://github.com/PX4/PX4-Autopilot/blob/main/Tools/setup/arch.sh). <!-- NEED px4_version -->
 
-The script installs (by default) all tools to build PX4 (without RTPS) for NuttX targets and run simulation with *jMAVsim*.
+The script installs (by default) all tools to build PX4 for NuttX targets and run simulation with *jMAVsim*.
```

Comment:
We might change this back to re-add "without DDS"

Reply:
@hamishwillee there are multiple ways to get micro-ROS and XRCE-dds:

1) Microxrcedds client + agent **built from source**: [xrcedds install](http://micro-xrce-dds.readthedocs.io/en/latest/installation.html) *you can install client and/or agent, as all of these packages are part of a superbuild cmake list, and will install dependencies for you!
2) Standalone microxrce-dds agent via snap install: [xrce agent standalone snap](https://snapcraft.io/micro-xrce-dds-agent)
3) Micro-ROS via ros2 colcon build [uros ros2 package:](https://github.com/micro-ROS/micro_ros_setup) * requires running a setup + build script using 'ros2 run' as well as the colcon build tools. 
4) micro-ROS via snap package: [uros snap](https://snapcraft.io/micro-ros-agent)

As TSC21 stated, fastdds is still a dependency, but the microxrcedds install from source also call out fastcdr among other packages (foonathon_memory, googletest, spdlog...). However, the **SuperBuild.cmake will attempt to find the package, if not found it will clone and build as a dependency**. The installation process is very simplified compared to the old docs.

Reply:
@TSC21"Are we targeting to use this without or with ROS2 only?"

Good question - isn't that up to you to answer? For RTPS docs we did include both cases.

1. My understanding is that both options are supported - right? 
2. My further assumption is that the vast majority of users will be accessing via ROS2. If so I suggest that we target these two cases completely separately, and start with instructions that assume ROS2 consumers. Does that make sense?
3. IN that case, what are the simplest instructions we can use to set up fastDDR, microROS and XRCE-dds _agent_ code? My assumption is that it is the [stuff in this gist](https://gist.github.com/julianoes/adbf76408663829cd9aed8d14c88fa29#micro-ros) down to the microROS topic - Can you confirm, or is there a better way? 
   > @dirksavage88 I think this is your option "3" above. 
4. My understanding is that you don't need to build the microROS or XRCE-DDS modules yourself directly, this just happens as part of the build. 
   - is that right? 
   - If you don't have FastDDS / ROS installed, does that just mean the module isn't built and included in firmware, but everything else will work?

Reply:
@dirksavage88 For the "lower priority"/non-ROS2 case what would you recommend for installation - i.e. what do you need? Do you know of any meaningful tutorials? For RTPS we had this listener: https://docs.px4.io/main/en/middleware/micrortps.html#creating-a-fast-dds-listener-application

Reply:
@hamishwillee not sure about 1. but I definitively agree on 2.
About 3. Assuming to use Foxy, Humble of Rolling as ROS2 distro (not Galactic) I would say that the simplest instructions are just cloning in the workspace the bare Agent library https://github.com/eProsima/Micro-XRCE-DDS-Agent/ and build it together with the other ros packages (px4_msgs in particular). The library will automatically fetch all its dependencies and in the end the agent will be started with just `MicroXRCEAgent ....`

Reply:
@beniaminopozzan Thanks! The reason I kind of thought using microROS installations as per tutorial might be good was future proofing. I.e. if we know that micro ROS contains XRCE-DDS agent and we expect micro ROS to be supported, then there is no particular cost in just following the micro ROS setup. Or am I being naïve? 

IN addition, I already have instructions to follow for micro ROS. That reduces my testing.

Of course if someone was to provide the step by step instructions for setup, things might be different :-)

Reply:
@hamishwillee 

> Note, @beniaminopozzan says that microROS client is not installed. So presumably this install approach works because microROS dependencies include the XRCE-DDS agent.

Micro ros client is "just" microdds client (the one that PX4 uses) for the low level communication (with the microdds agent) plus the ROS API to have ROS in the device https://micro.ros.org/docs/concepts/client_library/introduction/
the pipeline is something like
low-resource-device  -- microRosClient -- micro XRCE-DDS client ----- micro XRCE-DDS Agent  -- microRosAgent -- ROS2
Right now, PX4 directly connects to micro XRCE-DDS client.

> The reason I kind of thought using microROS installations as per tutorial might be good was future proofing. I.e. if we know that micro ROS contains XRCE-DDS agent and we expect micro ROS to be supported, then there is no particular cost in just following the micro ROS setup. Or am I being naïve?

No, it make sense!

Regrading the step by step instructions, if you are ok with a really bare-minimum version, they are like this:


Provided that you have ROS2 foxy, humble or rolling installed, let's assume humble (like here https://gist.github.com/julianoes/adbf76408663829cd9aed8d14c88fa29#ros2)

1. create a ros2 workspace
```sh
mkdir -p ~/px4_ros_ws/src
cd ~/px4_ros_ws/src
```
2. clone the dependencies `px4_msgs` and `micro XRCE-DDS Agent`
```sh
git clone https://github.com/PX4/px4_msgs.git
git clone https://github.com/eProsima/Micro-XRCE-DDS-Agent.git
```
3. Source the main environment and compile the workspace
```sh
cd ~/px4_ros_ws/
source /opt/ros/humble/setup.bash
colcon build
```
4. Source the compiled workspace and run the Agent (assuming a the default sitl scenario where the client tries to connect using udp4 on port 8888 like here https://gist.github.com/julianoes/adbf76408663829cd9aed8d14c88fa29#px4-sitl-with-gazebo-classic)
```sh
source install/local_setup.bash
MicroXRCEAgent udp4 -p 8888
```
at this point the tutorial can resume from here

> Leave it running, and open a new terminal for the next step.

https://gist.github.com/julianoes/adbf76408663829cd9aed8d14c88fa29#check-px4-topics

Reply:
Thanks very much. I think that's more than enough for me to start on this next week. Much appreciated.

Reply:
@beniaminopozzan Re your comment:

> Micro ros client is "just" microdds client (the one that PX4 uses) for the low level communication (with the microdds agent) plus the ROS API to have ROS in the device https://micro.ros.org/docs/concepts/client_library/introduction/
the pipeline is something like: 
> `low-resource-device -- microRosClient -- micro XRCE-DDS client ----- micro XRCE-DDS Agent -- microRosAgent -- ROS2`


A few follow on questions. 
1. I do understand that microROS allows ROS2 on the embedded controller. So it is a ROS2 API layer that connects to the micro XRCE-DDS client.
  - Why do you need a microROS agent? When we query our messages from ROS2 in the examples they recognise the ROS2 topics from uORB via XRCE-DDS, so what is the agent doing for us?

> Right now, PX4 directly connects to micro XRCE-DDS client.

Yes. But to be precise:
2. Does PX4 support/build the microROSClient in source?
3. Does PX4 include the client in firmware?

Basically I'm trying to understand "what happens next" - my assumption is that if 2 and 3 are "yes" then next step would be for someone to write a ros2 app on PX4, and that they could do that now. 

4. Is there a compelling use case for microROS. We generally have been pushing people to companion for years because the FC is resource constrained and harder to write for.

Reply:
> I do understand that microROS allows ROS2 on the embedded controller. So it is a ROS2 API layer that connects to the micro XRCE-DDS client.

Yes, I have the same understanding of microRO.S

> Why do you need a microROS agent?

I really don't know. Having all components under the "ROS2" hat is nice (I mean starting the Agent with `ros2 run micro_ros_agent micro_ros_agent`) but I guess there is something more, not strictly necessary, some other convenient feature maybe?

> Does PX4 support/build the microROSClient in source?
> Does PX4 include the client in firmware?

Not right now. Right now only micro XRCE-DDS Client is included in the firmware (in as a submodule of the [microdds](https://github.com/PX4/PX4-Autopilot/tree/main/src/modules/microdds_client) module).

> Basically I'm trying to understand "what happens next"

I can't help you here, let me ping @dagar for these aspects.

Reply:
What does it mean if `ros2 topic list` works, but when you try echo one of the displayed topics you get an error?

```
ubuntu@ubuntu-virtual-machine:~$ ros2 topic echo /fmu/out/vehicle_status
The message type 'px4_msgs/msg/VehicleStatus' is invalid
```

Reply:
I don't know if this applies anytime that error message appears, but in this case it means that ROS2 doesn't have the message definition for `px4_msgs/msg/VehicleStatus` (or that the message definition that it has is different from the one used to build that message, I have to check that). Did you not build the package `px4_msgs` o did you not source the associated environment?

Reply:
Thanks @beniaminopozzan - I think I'm a slow learner. Got it.